crates/symbolicator-service/src/services/symbolication/mod.rs

```diff
@@ -138,21 +144,44 @@ impl SymbolicationActor {
             .fetch_sources(self.objects.clone(), &stacktraces)
             .await;
 
-        let debug_sessions = module_lookup.prepare_debug_sessions();
+        // Map collected source contexts to frames and collect URLs for remote source links.
+        let mut remote_sources: HashMap<url::Url, Vec<&mut RawFrame>> = HashMap::new();
+        {
+            let debug_sessions = module_lookup.prepare_debug_sessions();
+
+            for trace in &mut stacktraces {
+                for frame in &mut trace.frames {
+                    if let Some(url) =
+                        module_lookup.try_set_source_context(&debug_sessions, &mut frame.raw)
+                    {
+                        // Only resolve source context from URLs if the frame is "in-app".
+                        if frame.raw.in_app.unwrap_or(false) {
```

Comment:
This also needs changes in the sentry processor to enable this, as it current does not send any `in_app` attributes.

The other problems with this that come to mind:
- `in_app` is being modified/set by grouping, which runs *after* symbolication.
- minidumps never set `in_app` at all. neither do some SDKs like Native. However those debug files do not (yet?) have support for source links.

I’m okay with merging this as is, but this might need to be revisited.

Reply:
in .NET, in_app is being sent from the SDK. What would need to change for that to get propagated here?